lib/main/atom/codefix/codefixProvider.ts

```diff
@@ -60,33 +45,12 @@ export class CodefixProvider implements IntentionsProvider {
       )
 
     const fixes = await Promise.all(requests)
-    const results: Intention[] = []
+    const results: ResultT[] = []
 
     for (const result of fixes) {
       if (result.body) {
         for (const fix of result.body) {
-          results.push({
-            priority: 100,
-            title: fix.description,
-            selected: () => {
-              fix.changes.forEach(async fix => {
-                const {buffer, isOpen} = await this.getTypescriptBuffer(fix.fileName)
-
-                buffer.buffer.transact(() => {
-                  for (const edit of fix.textChanges) {
-                    buffer.buffer.setTextInRange(spanToRange(edit), edit.newText)
-                  }
-                })
-
-                if (!isOpen) {
-                  buffer.buffer.save()
-                  buffer.on("saved", () => {
-                    buffer.buffer.destroy()
-                  })
-                }
-              })
-            },
-          })
+          results.push(formatCodeFix(fix))
```

Comment:
Since you don't need to await or do anything else with the result you get from `formatCodeFix(fix)`, this method could have a return type of `Promise<protocol.CodeAction[]>` and the callers can take care of mapping over the array. It would make it a bit simpler.

Reply:
Right.